### PR TITLE
Remove unused helm values

### PR DESCRIPTION
**Pull request by arkodg**

### CODE REVIEW

**Comment by arkodg** [comment]
hey @Xunzhuo should we disable metrics service and kube-rbac-proxy for 0.4.0 and enable it later in a future release ?

File: charts/gateway-helm/templates/envoy-gateway-deployment.yaml [hunk 1]
```diff
@@ -69,7 +69,7 @@ spec:
         - --v=0
         env:
         - name: KUBERNETES_CLUSTER_DOMAIN
-          value: {{ .Values.kubernetesClusterDomain }}
```
Comment (zirain) [inline charts/gateway-helm/templates/envoy-gateway-deployment.yaml:72]: why remove this?

Comment (arkodg) [reply charts/gateway-helm/templates/envoy-gateway-deployment.yaml:72]: Imo we need to treat helm values as APi fields and expose them when needed, else we cause breaking changes, the goal of this PR is to expose minimum helm values needed for 0.4.0

Comment (zirain) [reply charts/gateway-helm/templates/envoy-gateway-deployment.yaml:72]: but this need be configurable in the future, xref: https://github.com/envoyproxy/gateway/issues/1280

**Review by Xunzhuo — APPROVED** [review]

**Comment by Xunzhuo** [comment]
Thoughts on https://github.com/envoyproxy/gateway/pull/1277#discussion_r1162220737 ?
cc @arkodg, we can remove it for now

**Comment by Xunzhuo** [comment]
Merge it since I am adding docs for values

**Pull request merged**